### PR TITLE
MueLu: pass primal/dual interface node mapping via user data sublist

### LINKED ISSUE #7164: MueLu: How to pass primal/dual mapping to InterfaceMappingTransferFactory?
## Question
@trilinos/muelu @kliegeois 

I'm trying to understand how I can pass the information on the mapping of dual nodes to primal nodes for the `InterfaceMappingTransferFactory` introduced in PR #6931 for our application. I'm looking at the example in `packages/muelu/test/meshtying/MeshTyingBlocked_SimpleSmoother.cpp`.

As in the example, I want to read the preconditioner configuration from an xml-file. In contrast to the example, I want to create the preonditioner via `MueLu::CreateXpetraPreconditioner()`.

The necessary information is set on the parameter list as shown here: https://github.com/trilinos/Trilinos/blob/29bf4e7151cf8f0810b173a3291097a3322a9745/packages/muelu/test/meshtying/MeshTyingBlocked_SimpleSmoother.cpp#L211-L215

As far as I understand, this hard-codes the requires the sublist being named `myInterfaceAggs2`. However, the name of this list is defined in the xml-file as a user-specified string, so there's no guarantee that the user uses that particular name in every xml-file.

@kliegeois How can I pass the primal/dual node mapping to MueLu in a way, where I do not rely on particular sublist names in the MueLu xml-file?

### PR DESCRIPTION
**Pull request by mayrmt**
@trilinos/muelu 
@kliegeois 

## Motivation

This PR enables to pass user-specific data regarding the dual-to-primal node mapping required by the `InterfaceAggregationFactory` through the parameter list `"user data"` and removes the previous restriction, that the MueLu xml-file had to contain a very specific sublist with a hard-coded name.

## Related Issues

* Closes #7164 

## Stakeholder Feedback

None

## Testing

MueLu tests pass.

### CODE REVIEW

**Comment by mayrmt** [comment]
I can confirm, that these changes also help the user app, that originally triggered the underlying issue #7164.

File: packages/muelu/src/Utils/MueLu_Utilities_decl.hpp [hunk 1]
```diff
@@ -892,10 +892,34 @@ namespace MueLu {
 
 
 
-  /*! Removes the following non-serializable data (A,P,R,Nullspace,Coordinates) from level-specific sublists from inList
-    and moves it to nonSerialList.  Everything else is copied to serialList.  This function returns the level number of the highest level
-    for which non-serializable data was provided.
-    */
+  /*!
+  \bfief Extract non-serializable data from level-specific sublist and move to separate parameter list
```
Comment (kliegeois) [inline packages/muelu/src/Utils/MueLu_Utilities_decl.hpp:896]: bfief should be replaced by brief

level-specific sublist 
should be replaced by 
level-specific sublists

and potentially
 move to separate parameter list 
could be replaced by
 move it to a separate parameter list

File: packages/muelu/src/Utils/MueLu_Utilities_decl.hpp [hunk 2]
```diff
@@ -892,10 +892,34 @@ namespace MueLu {
...
+  - "K"
+  - Nullspace information "Nullspace"
+  - Coordinate information "Coordinates"
+  - "Node Comm"
+  - Primal-to-dual node mapping "DualNodeID2PrimalNodeID"
+  - "pcoarsen: element to node map
+
+  @param[in] inList List with all input parameters/data as provided by the user
+  @param[out] serialList All serializable data from the input list
+  @param[out] nonSerialList All non-serializable, i.e. rank-specific data form the input list
```
Comment (kliegeois) [inline packages/muelu/src/Utils/MueLu_Utilities_decl.hpp:918]: data form the input list 
should be replaced by 
data from the input list

**Comment by kliegeois** [comment]
It could be great if a MueLu developer can approve this PR too before merging.

**Comment by mayrmt** [comment]
I've removed the automerge label to wait for @trilinos/muelu approval before merging.

**Comment by mayrmt** [comment]
This PR mostly affects MueLu's `InterfaceAggregationFactory`, who's author has approved the changes. Other changes reformat documentation, but don't change functionality. Hence, I merge this now.

**Pull request merged**